xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailView.swift

```diff
@@ -169,8 +175,13 @@ struct MemoViewerDetailLoadedView: View {
             return .systemAction
         }
         
+        guard let did = user?.did else {
+            return .systemAction
```

Comment:
I don't think we want to default to system action here, since we know this will be a `sub://` URL.

Instead, we want to return `.handled` and log what went wrong.

However, I also think that we should be encoding everything we need to follow a URL into the URL. Let's consider adding the necessary DID to the URL.

BTW what role does the DID play in looking up the URL?

Reply:
> what role does the DID play in looking up the URL?

The DID is how we detect if we know this user by another (shorter) name. We may tap a link to `@ben.gordon.chris/hello` but if we follow `@ben` already we want to land on `@ben/hello`. If we don't pass the DID through then we have to call `resolve` on `@ben.gordon.chris`, introducing latency.